src/lib/storage/value_vector.hpp

```diff
@@ -0,0 +1,210 @@
+  void reserve(const size_t n);
+
+  // Reduce capacity to fit its size
+  void shrink_to_fit();
+
+  // Return a copy of the allocator object associated with the vector of values
+  PolymorphicAllocator<T> get_allocator();
+
+  // Return a reference to the underlying vector
+  const pmr_vector<T>& values() const;
+
+  // Return a pointer to the first element of the data vector
+  const T* data() const;
+
+  // Return the calculated size of ValueVector in main memory
+  size_t data_size() const;
+
+ protected:
+  pmr_vector<T> _values;
+
+  template <class Iter>
+  void _iterator_push_back(Iter first, Iter last) {
+    while (first != last) {
+      push_back(*first);
+      ++first;
+    }
+  }
+};
+
+template <>
+class ValueVector<FixedString> {
+ public:
+  explicit ValueVector(size_t string_length) : _string_length(string_length) {}
+
+  // Create a ValueVector of FixedStrings with given values by iterating over other container
+  template <class Iter>
+  ValueVector(Iter first, Iter last, size_t string_length) : _string_length(string_length) {
+    _iterator_push_back(first, last);
+  }
+
+  template <class Iter>
+  ValueVector(Iter first, Iter last) : _string_length(first->size()) {
+    _iterator_push_back(first, last);
+  }
+
+  // Create a ValueVector of FixedStrings with given values
+  ValueVector(const ValueVector<FixedString>&& other) : _string_length(other._string_length), _chars(other._chars) {}
+  ValueVector(const ValueVector<FixedString>& other) : _string_length(other._string_length), _chars(other._chars) {}
+  ValueVector(const ValueVector<FixedString>& other, const PolymorphicAllocator<size_t>& alloc)
+      : _string_length(other._string_length), _chars(other._chars, alloc) {}
+
+  // Add a string to the end of the vector
+  void push_back(const std::string& string);
+
+  // Return the value at a certain position.
+  const std::string operator[](const size_t n) const;
+
+  FixedString at(const ChunkOffset chunk_offset);
+
+  // TODO(team_btm): move??
+  class iterator : public boost::iterator_facade<iterator, FixedString, std::random_access_iterator_tag, FixedString> {
+   public:
+    iterator(size_t string_length, const pmr_vector<char>& vector, size_t pos = 0)
+        : _string_length(string_length), _chars(vector), _pos(pos) {}
+    iterator& operator=(const iterator& other) {
+      DebugAssert(_string_length == other._string_length && &_chars == &other._chars,
+                  "can't convert pointers from different vectors");
+      _pos = other._pos;
+      return *this;
+    }
+
+   private:
+    using facade = boost::iterator_facade<iterator, FixedString, std::random_access_iterator_tag, FixedString>;
+    friend class boost::iterator_core_access;
+    bool equal(iterator const& other) const { return this->_pos == other._pos; }
+    typename facade::difference_type distance_to(iterator const& other) const {
+      return (int64_t(other._pos) - int64_t(this->_pos)) / int64_t(_string_length);
```

Comment:
why int64_t? If this is about pointers, consider using [std::uintptr_t](http://en.cppreference.com/w/cpp/types/integer) for platform independence

(other opinions?)